### ISSUE TITLE
phpThumb / modPHPThumb fails on relative hosts domain

### ISSUE THREAD

**Issue by CyntaxDotDev** [modxbot/migrate#6891]
ReSpawN created Redmine issue ID 6891

We are creating a website for a new client which hasn't yet pointer their domain towards the new server. We currenty have that domain plus the IP (duh) in our local hosts file which works fine.

I wanted to create a new phpThumbOf image which failed epicly on all fronts. I lost about 1 hour of debugging when I eventually came to the conclusion that phpThumbOf (or perhaps even modPHPThumb) couldn't find the file due to the server not recognizing the file path. It was going to the wrong server since it didn't have the ip/domain reference in /etc/hosts. We nano'd it, added it and tadaa, dynamicly created path by phpThumbOf and it all works. 

Now, the strange thing, all other thumbnails like in the manager or when adding it to a TV (since it uses system/phpthumb.php) works fine. Perhaps it's going wrong somewhere between frontend dynamic files and backend hardcoded to a php file.

So the question remains, is this a bug or perfectly normal behaviour? A colleague of mine said that perhaps it would be more benneficial if phpThumbOf would resolve the path absolute (thus /home/usr/domains/domain.tld/public_html/ on a DirectAdmin installation).

Again, not sure it's a bug but it's a nice feature non the less.

**Comment by CyntaxDotDev** [comment]
ReSpawN submitted:

Eh, I found a strange event... when I add it to the hotlinking and offsite it works fine. Incidentally, images with spaces in them don't work either. Granted, images, filenames or even folders should never contain spaces, but not all clients understand that. 

Also, {http_host} is not dynamic. We've got clientname.tld and clientname.sofito.nl with the .sofito.nl domain being an alias. When we go to the alias domain, we get a small phpThumb hotlinking/offsite warning image and an empty site furthermore.

So to sum up;
{http_host} isn't dynamic per domain as it was in MODX 2.1.3-pl (tested, we've got like 20 domains linked to that specific MODX)
Spaces in URLs aren't compatible (anymore?)

**Comment by CyntaxDotDev** [comment]
ReSpawN submitted:

This bug keeps popping up more and more - still no solution...
